packages/react-scheduler/src/__tests__/ReactScheduler-test.js

```diff
@@ -208,6 +226,102 @@ describe('ReactScheduler', () => {
         expect(callbackLog).toEqual(['A0', 'B', 'A1']);
       });
     });
+
+    describe('when callbacks time out: ', () => {
+      // USEFUL INFO:
+      // startOfLatestFrame is a global that goes up every time rAF runs
+      // currentTime defaults to startOfLatestFrame inside rAF callback
+      // and currentTime defaults to 15 before next frame inside idleTick
+
+      describe('when there is no more time left in the frame', () => {
+        it('calls any callback which has timed out, waits for others', () => {
+          const {scheduleWork} = ReactScheduler;
+          startOfLatestFrame = 1000000000000;
+          currentTime = startOfLatestFrame - 10;
+          const callbackLog = [];
+          // simple case of one callback which times out, another that won't.
+          const callbackA = jest.fn(() => callbackLog.push('A'));
+          const callbackB = jest.fn(() => callbackLog.push('B'));
+          const callbackC = jest.fn(() => callbackLog.push('C'));
+
+          scheduleWork(callbackA); // won't time out
+          scheduleWork(callbackB, {timeout: 100}); // times out later
+          scheduleWork(callbackC, {timeout: 2}); // will time out fast
+
+          runRAFCallbacks(); // runs rAF callback
+          // push time ahead a bit so that we have no idle time
+          startOfLatestFrame += 16;
+          drainPostMessageQueue(); // runs postMessage callback, idleTick
```

Comment:
I plan to improve the API for writing these tests; instead of `drainPostMessageQueue` and `runRAFCallbacks` it should be more generic and descriptive, like so:

```
const shorterTimeout = 2;
const longerTimeout = 100;
scheduleWork(callbackA);
scheduleWork(callbackB, {timeout: longerTimeout});
scheduleWork(callbackC, {timeout: shorterTimeout});

// generic API for advancing a frame and configuring how much idle time there will be
advanceByOneFrame({timeRemaining: -1 * (longerTimeout * 2)});

// idle time was negative, so there was none and we passed the timeout threshhold for the shorterTimeout
expect(callbackLog).toEqual(['C']);

// ... etc.
```